train.py

```diff
@@ -1,15 +1,32 @@
-import argbind
+import time
 from datetime import datetime
 from pathlib import Path
-import time
-from torch.utils.data import DataLoader
-import torch.optim as optim
+
+import argbind
 import torch
+import torch.optim as optim
+from torch.utils.data import DataLoader
 from torch.utils.tensorboard import SummaryWriter
+from tqdm import tqdm
 
 import data
-from model import Scribe
 import utils
+from model import Scribe
+
+
+@argbind.bind
+def scribe_loss(prediction, target, target_mask, penup_weighting=.5):
+    mse_func = torch.nn.MSELoss(reduction='none')
+    binary_crossentropy_func = torch.nn.BCEWithLogitsLoss(reduction='none')
+
+    penup_loss = binary_crossentropy_func(prediction[:, :, 0:1], target[:, :, 0:1])
+    penup_loss = penup_loss.sum(dim=2)
+    coords_loss = mse_func(prediction[:, :, 1:3], target[:, :, 1:3])
+    coords_loss = coords_loss.sum(dim=2)
+    loss =  penup_weighting*penup_loss + (1-penup_weighting)*coords_loss
+    loss = loss.masked_select(target_mask)
```

Comment:
Interesting way of doing this! Where did you learn this Neil?
I didn't know about `masked_select`.

Reply:
Don't remember how I found out about it.  google searching, I imagine.